core/src/test/java/org/elasticsearch/node/internal/InternalSettingsPreparerTests.java

```diff
@@ -42,6 +44,8 @@
 import static org.hamcrest.Matchers.*;
 
 public class InternalSettingsPreparerTests extends ESTestCase {
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
```

Comment:
It is a matter of taste i suppose, but i still hate all junit support for exceptions like this.

The rule sucks because it has side effects, if we write another test that uses it, it can easily have some bogus leftover state from a previous test method?

Personally i still do it tests like this:

    try {
      something();
      fail("should have hit expected exception");
    } catch (SomeException expected) {
      assertTrue(expected.getMessage().contains("expected text"));
    }

This sucks too, in that its easy to forget the fail() part and have the whole test do nothing.